pallets/farming/src/gauge.rs

```diff
@@ -0,0 +1,494 @@
+            gauge_basic_rewards,
+            max_block,
+            gauge_amount: Default::default(),
+            total_time_factor: Default::default(),
+            gauge_last_block: current_block_number,
+            gauge_state: GaugeState::Bonded,
+        }
+    }
+}
+
+impl<T: Config> Pallet<T>
+where
+    BlockNumberFor<T>: AtLeast32BitUnsigned + Copy,
+    BalanceOf<T>: AtLeast32BitUnsigned + Copy,
+{
+    pub fn create_gauge_pool(
+        pid: PoolId,
+        pool_info: &mut PoolInfo<BalanceOf<T>, CurrencyIdOf<T>, AccountIdOf<T>, BlockNumberFor<T>>,
+        gauge_token: CurrencyIdOf<T>,
+        gauge_basic_rewards: BTreeMap<CurrencyIdOf<T>, BalanceOf<T>>,
+        max_block: BlockNumberFor<T>,
+    ) -> DispatchResult {
+        let gid = Self::gauge_pool_next_id();
+        pool_info.gauge = Some(gid);
+        let current_block_number = frame_system::Pallet::<T>::block_number();
+        let gauge_pool_info = GaugePoolInfo::new(
+            pid,
+            gauge_token,
+            pool_info.keeper.clone(),
+            pool_info.reward_issuer.clone(),
+            gauge_basic_rewards,
+            max_block,
+            current_block_number,
+        );
+
+        GaugePoolInfos::<T>::insert(gid, &gauge_pool_info);
+        GaugePoolNextId::<T>::mutate(|id| -> DispatchResult {
+            *id = id.checked_add(1).ok_or(ArithmeticError::Overflow)?;
+            Ok(())
+        })?;
+        Ok(())
+    }
+
+    pub fn gauge_add(
+        who: &AccountIdOf<T>,
+        gid: PoolId,
+        gauge_value: BalanceOf<T>,
+        gauge_block: BlockNumberFor<T>,
+    ) -> DispatchResult {
+        GaugePoolInfos::<T>::mutate(gid, |gauge_pool_info_old| -> DispatchResult {
+            if let Some(mut gauge_pool_info) = gauge_pool_info_old.take() {
+                let current_block_number = frame_system::Pallet::<T>::block_number();
+                gauge_pool_info.gauge_last_block = current_block_number;
+                gauge_pool_info.gauge_amount = gauge_pool_info
+                    .gauge_amount
+                    .checked_add(&gauge_value)
+                    .ok_or(ArithmeticError::Overflow)?;
+                let mut gauge_info =
+                    GaugeInfos::<T>::get(gid, who).unwrap_or_else(|| GaugeInfo::new(who.clone()));
+
+                ensure!(
+                    gauge_info.gauge_stop_block >= current_block_number
+                        || gauge_info.gauge_stop_block == Default::default(),
+                    Error::<T>::LastGaugeNotClaim
+                );
+
+                ensure!(
+                    gauge_pool_info.max_block
+                        >= gauge_info.gauge_stop_block - gauge_info.gauge_start_block + gauge_block,
+                    Error::<T>::GaugeMaxBlockOverflow
+                );
+
+                let incease_total_time_factor = if gauge_info.gauge_amount.is_zero() {
+                    gauge_info.gauge_stop_block = current_block_number;
```

Comment:
If `claimed_time_factor` still uses default `claimed_time_factor`, does it affect gauge claiming?
Because I found this field is not being set.